source/postprocess/geoid.cc

```diff
@@ -739,6 +739,125 @@ namespace aspect
+                      const double cos_component = sph_harm_vals.first; // real / cos part
+                      const double sin_component = sph_harm_vals.second; // imaginary / sine part
+
+                      // the conversion from geoid to gravity anomaly is given by gravity_anomaly = (l-1)*g/R_surface * geoid_anomaly
+                      // based on Forte (2007) equation [97]
+                      gravity_value += (geoid_coecos.at(ind)*cos_component+geoid_coesin.at(ind)*sin_component) * (ideg - 1) * surface_gravity / outer_radius * 1e5; // conversion to mgal
+                      ++ind;
+                    }
+                }
+              gravity_anomaly.push_back(gravity_value);
+            }
+
+          // Prepare the output data
+          if (output_in_lat_lon == true)
+            {
+              std::vector<std::pair<std::pair<double,double>,double> > stored_values_lon_lat;
+              double lon, lat;
+              for (unsigned int i=0; i<surface_cell_spherical_coordinates.size(); ++i)
+                {
+                  // Transfer the spherical coordinates to geographical coordinates
+                  lat = 90. - surface_cell_spherical_coordinates.at(i).first*(180./numbers::PI);
+                  lon = (surface_cell_spherical_coordinates.at(i).second <= numbers::PI
+                         ?
+                         surface_cell_spherical_coordinates.at(i).second*(180./numbers::PI)
+                         :
+                         surface_cell_spherical_coordinates.at(i).second*(180./numbers::PI) - 360.);
+                  stored_values_lon_lat.emplace_back(std::make_pair(lon,lat),gravity_anomaly.at(i));
+                }
+              // Write the solution to the stream output
+              for (unsigned int i=0; i<stored_values_lon_lat.size(); ++i)
+                {
+                  output_gravity_anomaly << stored_values_lon_lat.at(i).first.first
+                                         << ' '
+                                         << stored_values_lon_lat.at(i).first.second
+                                         << ' '
+                                         << stored_values_lon_lat.at(i).second
+                                         << std::endl;
+                }
+            }
+          else
+            {
+              std::vector<std::pair<Point<dim>,double> > stored_values_xyz;
+              for (unsigned int i=0; i<surface_cell_locations.size(); ++i)
+                {
+                  stored_values_xyz.push_back(std::make_pair(surface_cell_locations.at(i),gravity_anomaly.at(i)));
+                }
+              // Write the solution to the stream output
+              for (unsigned int i=0; i<stored_values_xyz.size(); ++i)
+                {
+                  output_gravity_anomaly << stored_values_xyz.at(i).first
```

Comment:
Same comment here, the loop above is not really necessay, you could do all that inside this loop and save the additional container.